### ISSUE TITLE
ignore defer calls

### ISSUE THREAD

**Issue by dmitris** [kisielk/errcheck#55]
currently errcheck finds a lot of calls like defer r.Body.Close() or defer db.Close().  Would it be possible to ignore them to make the tool's output more meaningful and get the real problems to better stand out?  Thanks!

**Comment by dmitris** [comment]
ok, let's close this

**Comment by nevet** [comment]
@dominikh could you explain why you choose to do `f.Close()` when `err` is already not nil?

**Comment by dominikh** [comment]
@nevet That code of mine looks wrong. It should always call f.Close. Whether it should overwrite an already non-nil error depends on the given requirements.

**Comment by afeld** [comment]
Here's what I ended up doing:

```go
func foo() error {
  f, err := os.Create("foo")
  if err != nil {
    return err
  }
  
  defer func() {
    err = f.Close()
    if err != nil {
      log.Fatalln(err)
    }
  }()
  // ...
}
```

**Comment by amnonbc** [comment]
A good suggestion. 
Lack of this option means you have to pollute your code with 
`defer func() { _ = r.Body.Close() }()`.

And without this it the Go standard library fails errcheck.

The authors of. this very useful tool have strong opionions on this issue and do not what to add this feature,
as is there right.

But I am going to fork the repo to https://github.com/amnonbc/errcheck
to add in a `-ignoredefer` option.
I'll post here when I have this working.

**Comment by amnonbc** [comment]
OK. I have hacked the code in my fork to ignore calls to`defer foo.Close()`

I would like to thank the authors of errcheck on writing an very useful tool,
and on writing clear, readable, idiomatic code. 
I enjoyed hacking the code, and it taught me a lot about AST tree traversal.

Anyone who wants to avoid littering their code with `defer func() { _ resp.Body.Close()}()`
is welcome to use my fork https://github.com/amnonbc/errcheck
